### PR TITLE
Fix RacerGetPrefixCol

### LINKED ISSUE #68: Issue with completion prefix
Maybe it changed recently in racer, but racer returns the byte position, not the column number when you call `racer prefix`.

So, the completion works nicely when you invoke it on the first line, but it doesn't on any other line :)

How to reproduce:

- Latest vim-racer
- Latest racer (``racer 2.0.2``)

Open a new buffer:

```
use std::rc::R<C-x><C-o>
use std::cell::R<C-x><C-o>
```

The first line will work perfectly. The second line will end-up being ``use std::cel::RRef`` (with ``completeopt=longest``).

Some solutions that I thought about:

1. Change ``s:RacerGetPrefixCol`` so it uses the completions to define the column position (so ``startpos`` would be ``col('.') - len(completions[0]) - 1``)
1. Propose a change on racer (an optional parameter to ``racer prefix`` to print the column number, maybe?)
1. Some hack to convert byte position to line + col

Thoughts?

### PR DESCRIPTION
**Pull request by fsouza**
`racer prefix` returns the byte position, not the column number.

I also defined b:tmpfname in RacerGetExpCompletions. The function relied
on this variable being defined elsewhere, which works, but is dangerous.

I don't know if this was a breaking change in racer. If this was a
breaking change in racer, should we care about backward compatibility?

Fix #68.

### CODE REVIEW

**Comment by tomprince** [comment]
I can confirm that vim-racer is broken without this, when using racer 2.0.3. The symptom appears to be that the completion prefix is also getting added as part of the completion.

File: plugin/racer.vim [hunk 1]
```diff
@@ -60,12 +60,13 @@ function! s:RacerGetPrefixCol(base)
...
     let prefixline = split(res, "\\n")[0]
-    let startcol = split(prefixline[7:], ",")[0]
-    return startcol
+    let startbyte = split(prefixline[7:], ",")[0]
+    return startbyte - line2byte(byte2line(startbyte)) + 1
 endfunction
 
 function! s:RacerGetExpCompletions(base)
     let col = col(".")-1
+    let b:tmpfname = tempname()
```
Comment (tomprince) [inline plugin/racer.vim:69]: This seem like an unrelated change?

Comment (fsouza) [reply plugin/racer.vim:69]: Yes, it's unrelated. I added a note in the commit message explaining this change too.

**Comment by Shougo** [comment]
Merged.
